pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConnectionPool.java

```diff
@@ -453,5 +453,8 @@ public void doMarkAndReleaseUselessConnections(){
         // Do release idle connections.
         releaseIdleConnectionTaskList.forEach(Runnable::run);
     }
-}
 
+    public ConcurrentMap<InetSocketAddress, ConcurrentMap<Integer, CompletableFuture<ClientCnx>>> getPool() {
+        return pool;
+    }
```

Comment:
We should return a new structure or immutable structure. Otherwise, the caller can modify the map which will introduce unpredictable behavior

My suggestion is we can change to `getConnections()` and only return a List structure.